webapp/src/lib/dialog-transaction.ts

```diff
@@ -72,16 +52,25 @@ export class DialogTransactionComponent implements OnInit {
         for (let i = -2; i <= 0; i++) {
             this.addBlock(this.blockIndex + i);
         }
-        const prog = (p: number, t: string) => DialogTransactionComponent.progress(this, p, t);
+        const prog = (p: number, t: string) => this.progress(p, t);
         try {
-            await this.data.worker(prog);
+            const result = await this.data.worker(prog);
             prog(-100, "Done");
+            clearTimeout(this.ubTimer);
+            this.blockIndex++;
+            this.addBlock();
+            setTimeout(() => {
+                this.dialogRef.close(result);
+            }, 1000);
         } catch (e) {
             this.error = e.toString();
         }
     }
 
     addBlock(index: number = this.blockIndex): Element {
+        if (!this.main) {
+            return null;
```

Comment:
Hum, so, `null` and `undefined`, I don't know why there is two bottom types and when to use each; from the typescript's [coding guidelines](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined), they prefers to use `undefined` so this is what I use. Do you have any preference/idea/limitation on each?

Reply:
Oh my - in your hands even typescript is safe!